### PR TITLE
Booflo compat but based and not cringe

### PR DESCRIPTION
**Pull request by abigailfails**
CompatEvents is gone and booflos implement IAgeableEntity, which as of the latest AC release and S&R beta make it compatible with potato poisoning and growth potions.

### CODE REVIEW

File: src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloBabyEntity.java [hunk 1]
```diff
@@ -369,6 +367,27 @@ public boolean onLivingFall(float distance, float damageMultiplier) {
...
+
+	@Override
+	public boolean canAge(boolean isGrowing) {
+		return isGrowing;
+	}
+
+	@Override
+	public LivingEntity attemptAging(boolean isGrowing) {
+		if (isGrowing) return growUp();
+		return this;
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloBabyEntity.java:388]:
```suggestion
		return isGrowing ? this.growUp() : this;
```

File: src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloAdolescentEntity.java [hunk 2]
```diff
@@ -568,6 +566,26 @@ public boolean canDespawn(double distanceToClosestPlayer) {
...
+	}
+
+	@Override
+	public boolean canAge(boolean isGrowing) {
+		return true;
+	}
+
+	@Override
+	public LivingEntity attemptAging(boolean isGrowing) {
+		return isGrowing ? growUp() : growDown();
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloAdolescentEntity.java:586]:
```suggestion
		return isGrowing ? this.growUp() : this.growDown();
```

File: src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloEntity.java [hunk 3]
```diff
@@ -1310,6 +1312,27 @@ public ItemStack getPickedResult(RayTraceResult target) {
...
+
+	@Override
+	public boolean canAge(boolean isGrowing) {
+		return !isGrowing;
+	}
+
+	@Override
+	public LivingEntity attemptAging(boolean isGrowing) {
+		if(!isGrowing) return growDown();
+		return this;
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloEntity.java:1333]:
```suggestion
		return isGrowing ? this : this.growDown();
```

**Review by SmellyModder — CHANGES_REQUESTED** [review]
Just some minor nitpicks.

File: src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloEntity.java [hunk 4]
```diff
@@ -1310,6 +1312,26 @@ public ItemStack getPickedResult(RayTraceResult target) {
...
 	}
 
+	@Override
+	public boolean hasGrowthProgress() {
+		return false;
+	}
+
+	@Override
+	public void resetGrowthProgress() {
+
```
Comment (SmellyModder) [inline src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloEntity.java:1322]:
```suggestion
```

**Review by SmellyModder — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloEntity.java (hunk 4 above)
Comment (abigailfails) [reply src/main/java/com/minecraftabnormals/endergetic/common/entities/booflo/BoofloEntity.java]: Wait, this is resolved?

**Comment by SmellyModder** [comment]
Just fixed it myself

**Review by SmellyModder — APPROVED** [review]

**Pull request merged**